app/controllers/patients_controller.rb

```diff
@@ -1,12 +1,23 @@
 class PatientsController < AdminController
   before_action :set_patient, only: [:update]
 
+  DEFAULT_PAGE_SIZE = 20
+
   def index
     authorize Patient, :index?
-    @patients = policy_scope(Patient)
-                               .not_contacted
-                               .order(device_created_at: :asc)
-                               .page(params[:page]).per(10)
+
+    @facility_id = params[:facility_id].present? ? params[:facility_id] : 'All'
+    selected_facilities = @facility_id == 'All' ? policy_scope(Facility.all) : policy_scope(Facility.where(id: @facility_id))
+    patients_to_show = policy_scope(Patient)
+                         .not_contacted
+                         .where(registration_facility: selected_facilities)
+
+    @per_page = params[:per_page].present? ? params[:per_page] : DEFAULT_PAGE_SIZE
+    per_page_count = @per_page == 'All' ? patients_to_show.size : @per_page.to_i
+    @patients = patients_to_show
+                  .order(device_created_at: :asc)
+                  .page(params[:page])
```

Comment:
What happens when params[:page] is `nil`?

Reply:
the `page` method and `params[:page]` are both provided by `kaminari`, and I'm not seeing a case where this happens.